File: android/src/main/kotlin/live/hms/hmssdk_flutter/hls_player/HMSHLSPlaybackStateExtension.kt
package live.hms.hmssdk_flutter.hls_player

import live.hms.hls_player.HmsHlsPlaybackState

class HMSHLSPlaybackStateExtension {
    companion object {
        fun toDictionary(hmsHlsPlaybackState: HmsHlsPlaybackState): HashMap<String, String> {
            val args = HashMap<String, String>()
            args["playback_state"] = when (hmsHlsPlaybackState) {
                HmsHlsPlaybackState.playing -> "playing"
                HmsHlsPlaybackState.stopped -> "stopped"
                HmsHlsPlaybackState.paused -> "paused"
                HmsHlsPlaybackState.buffering -> "buffering"
                HmsHlsPlaybackState.failed -> "failed"
                else -> "unknown"
            }
            return args
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/HLSPlayer/HMSHLSPlaybackStateExtension.swift
//
//  HMSHLSPlaybackStateExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 24/05/23.
//

import Foundation
import HMSHLSPlayerSDK

class HMSHLSPlaybackStateExtension {
    static func toDictionary(_ hmsHlsPlaybackState: HMSHLSPlaybackState) -> [String: String] {
        var args = [String: String]()
        switch hmsHlsPlaybackState {
        case .playing:
            args["playback_state"] = "playing"
        case .stopped:
            args["playback_state"] = "stopped"
        case .paused:
            args["playback_state"] = "paused"
        case .buffering:
            args["playback_state"] = "buffering"
        case .failed:
            args["playback_state"] = "failed"
        default:
            args["playback_state"] = "unknown"
        }
        return args
    }
}
